test/e2e/persistent_volumes.go

```diff
@@ -62,38 +77,63 @@ func nfsServerPodCleanup(c clientset.Interface, config VolumeTestConfig) {
 	}
 }
 
+// TODO Comment
+func pvPvcCleanup(c clientset.Interface, ns string, pv *api.PersistentVolume, pvc *api.PersistentVolumeClaim) {
+	if c != nil && len(ns) > 0 {
+		if pvc != nil && len (pvc.Name) > 0 {
+			deletePersistentVolumeClaim(c, pvc.Name, ns)
+		}
+		if pv != nil && len(pv.Name) > 0 {
+			deletePersistentVolume(c, pv.Name)
+		}
+	}
+}
+
 // Cleanup up pvs and pvcs in multi-pv-pvc test cases. All entries found in the pv and
 // claims maps are deleted.
 // Note: this is the only code that deletes PV objects.
-func pvPvcCleanup(c clientset.Interface, ns string, pvols pvmap, claims pvcmap) {
-
+func pvPvcMapCleanup(c clientset.Interface, ns string, pvols pvmap, claims pvcmap) {
 	if c != nil && len(ns) > 0 {
 		for pvcKey := range claims {
-			_, err := c.Core().PersistentVolumeClaims(pvcKey.Namespace).Get(pvcKey.Name)
-			if !apierrs.IsNotFound(err) {
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleting PVC %v ...", pvcKey)
-				err = c.Core().PersistentVolumeClaims(pvcKey.Namespace).Delete(pvcKey.Name, nil)
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleted  PVC %v", pvcKey)
-			}
+			deletePersistentVolumeClaim(c, pvcKey.Name, ns)
 			delete(claims, pvcKey)
 		}
 
-		for name := range pvols {
-			_, err := c.Core().PersistentVolumes().Get(name)
-			if !apierrs.IsNotFound(err) {
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleting PV %v ...", name)
-				err = c.Core().PersistentVolumes().Delete(name, nil)
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleted  PV %v", name)
-			}
-			delete(pvols, name)
+		for pvKey := range pvols {
+			deletePersistentVolume(c, pvKey)
+			delete(pvols, pvKey)
 		}
 	}
 }
 
+// Delete the PV.
+func deletePersistentVolume(c clientset.Interface, pvName string) {
+	if c != nil && len(pvName) > 0 {
+	        _, err := c.Core().PersistentVolumes().Get(pvName)
+        	if !apierrs.IsNotFound(err) {
+        		Expect(err).NotTo(HaveOccurred())
+        		framework.Logf("Deleting PersistentVolume %v", pvName)
+        		err := c.Core().PersistentVolumes().Delete(pvName, nil)
+        		Expect(err).NotTo(HaveOccurred())
+        	} else {
+			framework.Logf("", pvName)
+		}
+	}
+}
+
+// Delete the Claim
+func deletePersistentVolumeClaim(c clientset.Interface, pvcName string, ns string) {
+	if c != nil && len(pvcName) > 0 {
```

Comment:
The check for `c != nil and len(ns) > 0` is already done by the callers of this func; other than when directly called by the first _It()_ in the GCE framework. Maybe it's better to be safe than sorry but there is not really a performance concern for this test, so just a nit comment.